sycl/include/CL/sycl/detail/aligned_allocator.hpp

```diff
@@ -46,11 +45,16 @@ class aligned_allocator {
   pointer address(reference Val) const { return &Val; }
   const_pointer address(const_reference Val) { return &Val; }
 
-  // Allocate aligned (to Alignment) memory
+  // Allocate sufficiently aligned memory
   pointer allocate(size_t Size) {
-    Size += Alignment - Size % Alignment;
+    size_t NumBytes = Size * sizeof(value_type);
+    size_t Alignment =
```

Comment:
Suggest making Alignment "constexpr" or change the next line from NumBytes | (Alignment - 1)

Reply:
Replaced modulus with bitwise AND